### PR TITLE
Merge/develop 2 8

### PR DESCRIPTION
**Pull request by momchil-flex**
- [x] .github/workflows/sync-to-readthedocs-repo.yaml
- [x] docs/api/abstract_models.rst
- [x] docs/api/mediums.rst
- [x] docs/api/plugins/design.rst
- [x] pyproject.toml
- [x] tests/test_components/test_autograd.py
- [x] tests/test_components/test_geometry.py
- [x] tidy3d/components/field_projection.py
- [x] tidy3d/components/structure.py
- [x] tidy3d/plugins/autograd/README.md
- [x] tidy3d/plugins/dispersion/fit_fast.py
- [x] tidy3d/web/api/autograd/autograd.py

### CODE REVIEW

**Comment by tylerflex** [comment]
pushed the autograd conflict fixes

File: tidy3d/components/dispersion_fitter.py [hunk 1]
```diff
@@ -363,7 +363,7 @@ def get_default_weights(cls, eps: ArrayComplex1D) -> Tuple[float, float]:
     def pole_residue(self) -> Tuple[float, ArrayComplex1D, ArrayComplex1D]:
         """Parameters for pole-residue model in original units."""
         if self.eps_inf is None or self.poles is None:
-            return None
+            return PoleResidue(eps_inf=1, poles=[])
```
Comment (yaugenst-flex) [inline tidy3d/components/dispersion_fitter.py]: `PoleResidue` is not defined, but when imported at the module level causes a circular dependency. Could you double-check @caseyflex? See [here](https://github.com/flexcompute/tidy3d/actions/runs/11930870354/job/33252531826?pr=2079#step:6:22).
Probably needs a test too.

```
tidy3d/components/dispersion_fitter.py:366:20: F821 Undefined name `PoleResidue`
    |
364 |         """Parameters for pole-residue model in original units."""
365 |         if self.eps_inf is None or self.poles is None:
366 |             return PoleResidue(eps_inf=1, poles=[])
    |                    ^^^^^^^^^^^ F821
```

Comment (caseyflex) [reply tidy3d/components/dispersion_fitter.py]: sure! I couldn't test earlier with all the conflicts

Comment (caseyflex) [reply tidy3d/components/dispersion_fitter.py]: should be ok now, thanks for your work on the conflicts!

Comment (yaugenst-flex) [reply tidy3d/components/dispersion_fitter.py]: awesome, thanks! this is probably ready to go then @momchil-flex

**Pull request merged**